pkg/controller/migmigration/migmigration_controller.go

```diff
@@ -183,6 +183,9 @@ func (r *ReconcileMigMigration) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	// End staging conditions.
+	migration.Status.EndStagingConditions()
```

Comment:
I think you're on the right track, but this is just the first piece of the solution that needs to be implemented for this to work. 

We also need to move the `update CR` step up here as well.
```
err = r.Update(context.TODO(), migration)
```

And we also need to think about how moving the migration update will affect `MarkReconciled()`. I know the UI uses `MarkReconciled()` for some things, and we also use this to determine if we can skip validation steps in some cases. You'll need to investigate the migmigration controller and see if it's used here.

The other thing @jortel mentioned was a concern here was properly managing the conditions that we set within the `r.migrate()` call. I know we are setting a few conditions there. We need to make sure that moving the `EndStagingConditions` and `update()` doesn't cause condition flicker.

Reply:
Just to re-share the re-arrangement @jortel proposed would work here:


```
Potentially:
• begin staging
• validate etc
• end staging
• update CR
• migrate()

The migrate only manages a few conditions (Running, Failed, Succeeded) that would 
need to be managed outside if staging which means explicitly managed using 
SetCondition() DeleteCondition().  The predicate would protect against any echo reconciles.
```